src/utils/file_manager.cpp

```diff
@@ -0,0 +1,68 @@
+//
+// Created by blinku on 15.12.18.
+//
+
+#include "file_manager.h"
+
+#include "pch.h"
+#include <fstream>
+#include <sys/mman.h>
+
+namespace Bald::Utils {
+
+    std::string FileManager::ReadFile(const char *filePath, Size size) {
+        if (size == SMALL)
+            return ReadSmallFile(filePath);
+        return ReadBigFile(filePath);
+    }
+
+    std::string FileManager::ReadSmallFile(const char *filePath) {
+        FILE *file = fopen(filePath, "r");
+
+        if (!file) {
+            CORE_LOG_WARN("Couldn't open the file at path: " + static_cast<std::string>(filePath));
+            return std::string("Error!");
+        }
+
+        fseek(file, 0, SEEK_END);
+        auto stringSize = static_cast<unsigned long>(ftell(file));
+        fseek(file, 0, SEEK_SET);
+
+        auto *buffer = new char[stringSize + 1];
```

Comment:
What happens when we do not have memory and buffer == null? Add check for that.

Reply:
operator new throws a std::bad_alloc exception on failure, unless you're explicitly using the nothrow version